### ISSUE TITLE
Resume context in Chrome >= 71

### ISSUE THREAD

**Issue by thijstriemstra** [katspaugh/wavesurfer.js#1531]
The following warning appears in Chrome >= 71 on wavesurfer-js.org:

```
webaudio.js:76 The AudioContext was not allowed to start.
It must be resumed (or created) after a user gesture on the page.
```

https://goo.gl/7K7WLu

**Comment by nsomaru** [comment]
Is there any solution or workaround to this? Works fine in Firefox but refuses to play in Chrome.

**Comment by thijstriemstra** [comment]
Yeah, see the link; https://goo.gl/7K7WLu Basically create an AudioContext on a user gesture, like button press, and pass that wavesurfer using `audioContext` parameter.

**Comment by sundayz** [comment]
Any idea why Chrome emits this warning when creating the web audio audiocontext?
![image](https://user-images.githubusercontent.com/11685806/88923070-0675bd80-d269-11ea-95a1-49944d794bba.png)

Wavesurfer isn't set to autoplay... there's no audio playing on my page at startup. It always waits for the user to click buttons. But now even if you try to call wavesurfer.play() after a button press it won't start anymore.

Is it enough to call .resume() on the audiocontext before wavesurfer.play()? Or do I have to re-create the AudioContext object?

To make matters worse, the warning doesn't show on localhost, but it shows on the live website.

**Comment by marizuccara** [comment]
Hi, I do this and I have not any warning on console.

![image](https://user-images.githubusercontent.com/44341274/88925569-fa8ff880-d274-11ea-9f5a-f6721eb9cd8b.png)

**Comment by thijstriemstra** [comment]
@sundayz did you check https://goo.gl/7K7WLu?

**Comment by sundayz** [comment]
Thanks guys. Calling wavesurfer.backend.ac.resume() myself works. I'm not sure why it doesn't work without me doing it, since if I look at the code inside WebAudio.play(), WaveSurfer should be calling resume() for me.

**Comment by marizuccara** [comment]
Maybe is better to do it from the outside, because when the Promise from the resuming of the AudioContext is returned, you could want do do more stuffs, instead of only call play.

**Comment by sundayz** [comment]
https://github.com/katspaugh/wavesurfer.js/blob/master/src/webaudio.js#L684

It is resumed here but doesn’t seem to work 🤔 maybe because it’s inside a Promise...?

**Comment by marizuccara** [comment]
It should play only when audioContext is resumed, that is, when the Promise returned from resume method  is solved

**Comment by em-rad** [comment]
I've gotten the waveform to show and get rid of Chrome's (Version 84.0.4147.105) autoplay error when:

1. I have a button to handle `wavesurfer.playPause()`
2. The waveform div is `position:relative`

**Comment by X-Raym** [comment]
Same issue there. Can't this be fixed from wavesurfer.js side ?

**Comment by sundayz** [comment]
I think Chrome wants resume to be called directly in the click handler so it knows for sure the user is the one who started the audio, but play() is a Promise, so resume is not called inside the click handler. This is just a theory.

**Comment by meghagoga0617** [comment]
I have a very simple script to test , but I cannot play .
Till now I have tried many things as per above guidance none work, any idea ?

<!doctype html>
<html lang="en">
 <head>
  <meta charset="UTF-8">
  <meta name="Generator" content="EditPlus®">
  <meta name="Author" content="">
  <meta name="Keywords" content="">
  <meta name="Description" content="">
 
<script src="https://unpkg.com/wavesurfer.js"></script>


  <title>Document</title>
 </head>
 <body>
  <div id="waveform" style="position:relative"></div>
  <div class="controls" style="text-align:center;">
						<button class="playin" onclick="plays()">			
						Play
						/
						Pause
						</button>					
					</div>
 </body>

 <script type="text/javascript">
 /*window.AudioContext = window.AudioContext||window.webkitAudioContext;
var context = new AudioContext();

if(context.state === 'suspended')
  alert("auto-play failed!!");*/


/*wavesurfer.current.backend.ac.resume().then(() => {
	wavesurfer.current.play()
	.then(()=>setPlaying(true))
	.catch(()=>playin && setPlaying(false));
});*/
var wavesurfer = WaveSurfer.create
	({
    container: '#waveform',
    waveColor: 'violet',
    progressColor: 'purple'
	 
	});
function plays(){
	
	wavesurfer.load('song2.wav');
	wavesurfer.on('ready', function () {
	wavesurfer.backend.ac.resume();
    wavesurfer.play();
	});
}
 </script>
</html>

**Comment by X-Raym** [comment]
@katspaugh is this solved?

**Comment by katspaugh** [comment]
Reopened.

**Comment by entonbiba** [comment]
seems chrome is following ios safari method as well now,

**Comment by katspaugh** [comment]
Closing because in v7 there's no web audio backend.

**Comment by X-Raym** [comment]
@katspaugh webaudio backend was mostmy used to be able to not wait for full sound to be downloaded to press play in the middle of it (or start). Is this avlaible without it now ?

**Comment by katspaugh** [comment]
It downloads the complete audio file in both cases. One positive difference from v6 though is that the media element receives the same downloaded data as source, so the file is fetched only once. Please see https://github.com/katspaugh/wavesurfer.js/discussions/2762 for more details.

**Comment by X-Raym** [comment]
@katspaugh oh so it will be full HTML5 MediaElement... Interesting, I'll keep on eye on v7 official, especially cause it seems to fix old bugs like double downloads. Thx!

**Comment by X-Raym** [comment]
Hi, I still have the issue, in wavesurfer 7.0.6,

Here is a stripped down version on my init.js, https://gist.github.com/X-Raym/4ac1a331adf5ef3a126e0c3ea4bee240

I don't see what is wrong.

In Firefox also there is a console log message: in french:

```
Un AudioContext n’a pas pu démarrer automatiquement. Il doit être créé ou repris après un geste de l’utilisateur sur la page.
```

and in Brave (chrome):

```
wavesurfer_7.0.6.min.js The AudioContext was not allowed to start. It must be resumed (or created) after a user gesture on the page. https://goo.gl/7K7WLu
```

How to fix this?

Thx!

**Comment by X-Raym** [comment]
Note: brave traceroute goes up to this function in the minified version of wavesurfer 7.0.6

```js
    "function" == typeof SuppressedError && SuppressedError;
    const e = {
        decode: function(e, i) {
            return t(this, void 0, void 0, (function*() {
                const t = new AudioContext({⚠
                    sampleRate: i
                })
                  , s = t.decodeAudioData(e);
                return s.finally((()=>t.close())),
                s
            }
            ))
        },
        ```

**Comment by katspaugh** [comment]
@X-Raym thanks, I'll open a new issue for this.

**Comment by thijstriemstra** [comment]
> Closing because in v7 there's no web audio backend.

This has nothing to do with that.

> thanks, I'll open a new issue for this.

Why is this ticket not sufficient?

**Comment by katspaugh** [comment]
@thijstriemstra thanks for your input. I’ve made a new issue to not disturb you with notifications on this one.
